### PR TITLE
CI: add support for github actions

### PR DESCRIPTION
**Pull request by bassosimone**
When working on https://github.com/ooni/probe-engine/issues/893, I was
proceeding slower than I could because I had not enough confidence that
I was not breaking stuff. So, I'd say it's time to enable some form of
CI for this project. I also see that there is already a Travis file but
I also see that Travis is not running. Since we are already using GHA
for a bunch of client side projects, this strikes me as a good time to
switch this repository to using GHA.

Hopefully everything will work out of the box. If not, then we will
need to spend more time debugging after this first commit.

When everything is WAI, I'll purge .travis.yml.

### CODE REVIEW

**Comment by bassosimone** [comment]
So, the issue seems to be the version of the tools installed.

**Comment by bassosimone** [comment]
## Interesting findings

We were pinning to a version of the NDK that was not the latest version. This was causing the CI to fail because such an NDK was not available at the CI server. I initially thought this was an issue of the GHA runner, but it was our Gradle config.

See https://github.com/ooni/probe-android/pull/364/commits/fd6ffdfc3a718e262e58aa442e781d24cf49fd4f

**Comment by bassosimone** [comment]
## Tests are broken

It seems we only have minor breakages but here they are:

```
> Task :app:connectedDevFullDebugAndroidTest

org.openobservatory.ooniprobe.AutomateScreenshotsTest > testTakeScreenshot[test(AVD) - 10] FAILED 
	androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: with id: org.openobservatory.ooniprobe.dev:id/recyclerView
	

org.openobservatory.ooniprobe.common.ApplicationTest > packageName[test(AVD) - 10] FAILED 
	org.junit.ComparisonFailure: expected:<...bservatory.ooniprobe[]> but was:<...bservatory.ooniprobe[.dev]>
	at org.junit.Assert.assertEquals(Assert.java:115)

org.openobservatory.ooniprobe.common.ApplicationTest > packageName[test(AVD) - 10] FAILED 
	org.junit.ComparisonFailure: expected:<...bservatory.ooniprobe[]> but was:<...bservatory.ooniprobe[.dev]>
	at org.junit.Assert.assertEquals(Assert.java:115)

> Task :app:connectedDevFullDebugAndroidTest FAILED
```

See https://github.com/ooni/probe-android/runs/1101236904#step:3:329

**Comment by bassosimone** [comment]
So, it seems that running the correct test command "fixes" the tests. I'll further investigate later.

**Comment by bassosimone** [comment]
A check is failing:

```
> Task :app:connectedStableFullDebugAndroidTest
Starting 62 tests on test(AVD) - 10

org.openobservatory.ooniprobe.client.OONIAPIClientTest > getMeasurementSuccess[test(AVD) - 10] FAILED 
	java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
Tests on test(AVD) - 10 failed: Instrumentation run failed due to 'Process crashed.'

> Task :app:connectedStableFullDebugAndroidTest FAILED
```

See https://github.com/ooni/probe-android/runs/1101443946#step:3:281.

This problem seems to be related to replication delay between two different data centres causing the database to be unable to service normal requests, an issue that has been pestering us on and off for quite some time now. Because we are about to migrate off the data center containing the database anyway, there is little action we can take at this time.

Because of that, I'd say it's safe for us to move forward with the review of this PR.

File: app/src/androidTest/java/org/openobservatory/ooniprobe/client/OONIAPIClientTest.java [hunk 1]
```diff
@@ -23,7 +23,6 @@
 
 public class OONIAPIClientTest extends AbstractTest {
     private static final String EXISTING_REPORT_ID = "20190113T202156Z_AS327931_CgoC3KbgM6zKajvIIt1AxxybJ1HbjwwWJjsJnlxy9rpcGY54VH";
-    private static final String EXISTING_REPORT_ID_2 = "20190702T000027Z_AS5413_6FT78sjp5qnESDVWlFlm6bfxxwOEqR08ySAwigTF6C8PFCbMsM";
```
Comment (lorenzoPrimi) [inline app/src/androidTest/java/org/openobservatory/ooniprobe/client/OONIAPIClientTest.java:26]: 👏

**Comment by lorenzoPrimi** [comment]
🐳

**Pull request merged**